test/e2e/tests/metrics/delete-metametrics-data.spec.ts

```diff
@@ -0,0 +1,261 @@
+        return {
+          statusCode: 200,
+        };
+      }),
+    await mockServer
+      .forPost('https://metametrics.metamask.test/regulations/sources/test')
+      .withHeaders({ 'Content-Type': 'application/vnd.segment.v1+json' })
+      .withBodyIncluding(
+        JSON.stringify({
+          regulationType: 'DELETE_ONLY',
+          subjectType: 'USER_ID',
+          subjectIds: ['fake-metrics-id'],
+        }),
+      )
+      .thenCallback(() => ({
+        statusCode: 200,
+        json: { data: { regulateId: 'fake-delete-regulation-id' } },
+      })),
+    await mockServer
+      .forGet(
+        'https://metametrics.metamask.test/regulations/fake-delete-regulation-id',
+      )
+      .withHeaders({ 'Content-Type': 'application/vnd.segment.v1+json' })
+      .thenCallback(() => ({
+        statusCode: 200,
+        json: {
+          data: {
+            regulation: {
+              overallStatus: 'FINISHED',
+            },
+          },
+        },
+      })),
+  ];
+};
+/**
+ * Scenarios:
+ * 1. Deletion while Metrics is Opted in.
+ * 2. Deletion while Metrics is Opted out.
+ * 3. Deletion when user never opted for metrics.
+ */
+describe('Delete MetaMetrics Data @no-mmi', function (this: Suite) {
+  it('while user has opted in for metrics tracking', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder()
+          .withMetaMetricsController({
+            metaMetricsId: 'fake-metrics-id',
+            participateInMetaMetrics: true,
+          })
+          .build(),
+        defaultGanacheOptions,
+        title: this.test?.fullTitle(),
+        testSpecificMock: mockSegment,
+      },
+      async ({
+        driver,
+        mockedEndpoint: mockedEndpoints,
+      }: TestSuiteArguments) => {
+        await unlockWallet(driver);
+
+        await driver.clickElement(selectors.accountOptionsMenuButton);
+        await driver.clickElement(selectors.globalMenuSettingsButton);
+        await driver.clickElement(selectors.securityAndPrivacySettings);
+
+        await driver.findElement(selectors.deletMetaMetricsSettings);
+        await driver.clickElement(selectors.deleteMetaMetricsDataButton);
+
+        // there is a race condition, where we need to wait before clicking clear button otherwise an error is thrown in the background
+        // we cannot wait for a UI conditon, so we a delay to mitigate this until another solution is found
+        await driver.delay(3000);
+        await driver.clickElementAndWaitToDisappear(selectors.clearButton);
+
+        const deleteMetaMetricsDataButton = await driver.findElement(
+          selectors.deleteMetaMetricsDataButton,
+        );
+        await (
+          deleteMetaMetricsDataButton as WebElementWithWaitForElementState
+        ).waitForElementState('disabled');
+        assert.equal(
+          await deleteMetaMetricsDataButton.isEnabled(),
+          false,
+          'Delete MetaMetrics data button is disabled',
+        );
```

Comment:
Question: it seems like both assertions are waiting for button to be disabled, can we keep only one?